File: kotlin-swift-interopedia-samples-without-skie/iosApp/iosApp/Classes/EnumClass.swift
import Foundation
import shared

func useEnumClass() {
    let e1 = EnumClass.entryOne
    let _ = EnumClass.entryTwo

    let _ = EnumClass.entryOne.name
    let _ = EnumClass.entryOne.type

    let _ = EnumClass.companion.findByType(type: "entry_two")
    let optionalResult = EnumClass.companion.findByType(type: "entry_two_trheee")

    print("Enum example: \(e1)")
    print("Switching: \(switchEnumClass(enumClassExample: e1))")
    print("Non-existing entry: \(String(describing: optionalResult))")
}

private func switchEnumClass(enumClassExample: EnumClass) -> String {
    switch enumClassExample {
    case .entryOne: return "entryOne"
    case .entryTwo: return "entryTwo"
    default: return "default"
    }
}

File: kotlin-swift-interopedia-samples-without-skie/shared/src/commonMain/kotlin/com/jetbrains/swiftinteropplayground/classes/EnumClass.kt
package com.jetbrains.swiftinteropplayground.classes

enum class EnumClass(val type: String) {

    ENTRY_ONE("entry_one"),
    ENTRY_TWO("entry_two");

    companion object {
        fun findByType(type: String) = values().find { it.type == type }
    }

}